x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_actions/actions/delete/confirm_delete_endpoint/index.tsx

```diff
@@ -5,19 +5,63 @@
  * 2.0.
  */
 
-import React from 'react';
-import { EuiConfirmModal } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { EuiButtonEmpty, EuiConfirmModal, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { euiThemeVars } from '@kbn/ui-theme';
+
 import * as i18n from './translations';
+import { useScanUsage } from '../../../../../../../hooks/use_scan_usage';
+import { InferenceEndpointUI, InferenceUsageInfo } from '../../../../../types';
+import { RenderMessageWithIcon } from '../../component/render_message_with_icon';
+import { ScanUsageResults } from '../../component/scan_usage_results';
 
 interface ConfirmDeleteEndpointModalProps {
   onCancel: () => void;
   onConfirm: () => void;
+  inferenceEndpoint: InferenceEndpointUI;
 }
 
 export const ConfirmDeleteEndpointModal: React.FC<ConfirmDeleteEndpointModalProps> = ({
   onCancel,
   onConfirm,
+  inferenceEndpoint,
 }) => {
+  const [isFetching, setIsFetching] = useState<boolean>(true);
+  const [listOfUsages, setListOfUsages] = useState<InferenceUsageInfo[]>([]);
+  const [deleteDisabled, setDeleteDisabled] = useState<boolean>(true);
+  const [ignoreWarningCheckbox, setIgnoreWarningCheckbox] = useState<boolean>(false);
+
+  const { data } = useScanUsage({
+    type: inferenceEndpoint.type,
+    id: inferenceEndpoint.endpoint,
+  });
+
+  const onCheckboxChange = (state: boolean) => {
+    setIgnoreWarningCheckbox(state);
+    if (state) {
+      setDeleteDisabled(false);
+    } else {
+      setDeleteDisabled(true);
+    }
+  };
+
+  useEffect(() => {
+    if (!data) return;
+    setIsFetching(false);
+
+    const indices = data.indexes.map((index, id) => ({ label: index, type: 'Index' }));
```

Comment:
nonblocking - nitpick: unless reading this line, it was confusing to see `label` is used for navigation etc. it is the id, which is used as label as well. in some place.